src/el.ts

```diff
@@ -1,9 +1,19 @@
+interface EventHandler<T, E extends Event> {
+  (
+    e: E & {
+      currentTarget: T;
+      target: HTMLElement;
```

Comment:
Not sure why, but solidjs defines its own `DOMElement` type here instead of `HTMLElement`. I just took another look at it, and it turns out it just resolves to `Element`. (I also don't know the difference between `Element` and `Node`, but if we can trust their implementation I guess this should be an `Element`?) https://github.com/ryansolid/dom-expressions/blob/main/packages/dom-expressions/src/jsx.d.ts#L9

I wonder if the reason for that is because it doesn't necessarily have to be HTML, but could be an SVG or something. So, `HTMLElement` might technically be wrong in this case.